daemon/endpoint.go

```diff
@@ -213,6 +213,14 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 
 	ep.UpdateLabels(d, addLabels, infoLabels, true)
 
+	// Do not add endpoint to the endpoint manager if client abort connection
+	select {
+	case <-ctx.Done():
+		log.WithError(ctx.Err()).Warn("Aborting endpoint join due client connection closed")
+		return PutEndpointIDFailedCode, fmt.Errorf("aborting endpoint %d join due client connection closed", ep.ID)
+	default:
+	}
+
```

Comment:
What happens if a client closes the connection after `AddEndpoint` has successfully returned?

Wouldn't it make sense to do a cleanup at the end of the method `createEndpoint` if we detect that the connection has been closed?

Reply:
That should be covered by lines 290-297 of this file.